### DISCUSSION TITLE
Error "vm" undeclared identifier in lesson 7

### DISCUSSION THREAD

**Discussion by aaryachamkeri** [Cyfrin/foundry-full-course-cu#637] [General]
I was on the lesson about scripts in foundry and deploying scripts, and when I tried to run my deploy script using the following command: `forge script script/DeploySimpleStorage.s.sol ` I am getting the following error:
```
Compiler run failed:
Error (7576): Undeclared identifier.
  --> script/DeploySimpleStorage.s.sol:11:9:
   |
11 |         vm.startBroadcast();
   |         ^^

Error (7576): Undeclared identifier.
  --> script/DeploySimpleStorage.s.sol:13:9:
   |
13 |         vm.stopBroadcast();
   |         ^^
```

This is my code for the script:
```
// SPDX-License-Identifier: MIT

pragma solidity ^0.8.18;

import {Script} from "forge-std/Script.sol";
import {SimpleStorage} from "../src/SimpleStorage.sol";

contract DeploySimpleStorage {
    function run() external returns (SimpleStorage) {
        vm.startBroadcast();
        SimpleStorage simpleStorage = new SimpleStorage();
        vm.stopBroadcast();
        return simpleStorage;
    }
}
```
How do I fix this?

**Comment by theirrationalone** [comment]
You forgot to inherit `Script` into your child `contract`

Do this:

```soldity
contract DeploySimpleStorage is Script {
    //statements...
}
```

**Reply by aaryachamkeri** [reply]
Holy damn I am stupid thank you.

**Reply by theirrationalone** [reply]
Not a stupidity, But just an omission that we all went through at initial stages of development! 😉

We have our COMMUNITY. 

NEVER DISCOURAGE YOURSELF!

**Comment by theirrationalone** [comment]
Heyy @MehreenMunsifAli !, You just need to hit `Enter Key` in between `{}` Which is followed by `is Script` and need to put that `run` function and all other future code code snippets inside it.

;`)

**Reply by MehreenMunsifAli** [reply]
Thanks a lot @theirrationalone 
I finally found my mistake after spending 2 hours.

**Reply by alfheimrShiven** [reply]
Hey @MehreenMunsifAli do mark @theirrationalone solution that worked as the answer to this thread and close. Cheers!
